genkit-tools/common/src/eval/evaluate.ts

```diff
@@ -113,54 +158,72 @@ export async function getMatchingEvaluators(
       allEvaluatorActions.push(allActions[key]);
     }
   }
-  const filteredEvaluatorActions = allEvaluatorActions.filter(
-    (action) => !evaluators || evaluators.split(',').includes(action.name)
+  return allEvaluatorActions;
+}
+
+export async function getMatchingEvaluatorActions(
+  runner: Runner,
+  evaluators?: string[]
+): Promise<Action[]> {
+  if (!evaluators) {
+    return [];
+  }
+  const allEvaluatorActions = await getAllEvaluatorActions(runner);
+  const filteredEvaluatorActions = allEvaluatorActions.filter((action) =>
+    evaluators.includes(action.name)
   );
   if (filteredEvaluatorActions.length === 0) {
     if (allEvaluatorActions.length == 0) {
       throw new Error('No evaluators installed');
-    } else {
-      const availableActions = allEvaluatorActions
-        .map((action) => action.name)
-        .join(',');
-      throw new Error(
-        `No evaluators matched your specifed filter: ${evaluators}. All available evaluators: '${availableActions}'`
-      );
     }
   }
   return filteredEvaluatorActions;
 }
 
-async function bulkRunAction(
-  runner: Runner,
-  actionRef: string,
-  inputs: any[],
-  auth?: string
-): Promise<RunActionResponse[]> {
-  let responses: RunActionResponse[] = [];
+async function bulkRunAction(params: {
+  runner: Runner;
+  actionRef: string;
+  inputs: any[];
+  auth?: string;
+}): Promise<BulkRunResponse[]> {
+  const { runner, actionRef, inputs, auth } = params;
+  let responses: BulkRunResponse[] = [];
   for (const d of inputs) {
     logger.info(`Running '${actionRef}' ...`);
-    let response = await runner.runAction({
-      key: actionRef,
-      input: {
-        start: {
-          input: d,
+    let response: BulkRunResponse;
+    try {
+      const runActionResponse = await runner.runAction({
+        key: actionRef,
+        input: {
+          start: {
+            input: d,
+          },
+          auth: auth ? JSON.parse(auth) : undefined,
         },
-        auth: auth ? JSON.parse(auth) : undefined,
-      },
-    });
+      });
+      response = {
+        traceId: runActionResponse.telemetry?.traceId,
+        response: runActionResponse.result,
+      };
+    } catch (e: any) {
+      const traceId = e?.data?.details?.traceId;
+      response = { traceId };
+    }
     responses.push(response);
   }
 
   return responses;
 }
 
-async function fetchDataSet(
-  runner: Runner,
-  flowName: string,
-  states: BulkRunResponse[],
-  parsedData: EvalFlowInput
-): Promise<EvalInput[]> {
+async function fetchDataSet(params: {
```

Comment:
nit: can we rename this to fetchEvalInput to distinguish from the stored datasets?